src/pri-dual-vec.md

```diff
@@ -0,0 +1,193 @@
+# Primal and Dual Vectors
+
+The finite element method uses vectors of data in a variety of ways and the
+differences can be subtle.  MFEM defines `GridFunction`, `LinearForm`, and
+`Vector` classes which help to distinguish the different roles that vectors of
+data can play.
+
+<center>**Graphical summary of Primal, Dual, DoF, True DoF vectors**</center>
+![](img/FEM-LA-Diagrams.png)
+
+## Primal Vectors
+
+The finite element method is based on the notion that a smooth function can be
+approximated by a sum of piece-wise functions called *basis functions*:
+$$f(\vec{x})\approx\sum_i f_i \phi_i(\vec{x}) \label{expan}$$
+The support of an individual basis function, $\;\phi_i(\vec{x})$, will either be
+a single zone or a collection of zones that share a common vertex, edge, or
+face.  The expansion coefficients, $\;f_i$, are linear functionals of the field
+being approximated, $\;f(\vec{x})$ in this case.  The $\;f_i$ could be as simple
+as values of the function at particular points, called interpolation points,
+e.g. $\;f_i=f(\vec{x}\_i)$ or they could be integrals of the field over
+submanifolds of the domain, e.g. $\;f_i = \int_{\Omega_i}f(\vec{x})d\vec{x}$.
+There are many possibilities but the expansion coefficients must be linear
+functionals of $\;f(\vec{x})$.  The expansion coefficients are often called
+*degrees of freedom*, or *DoFs* for short, though in certain cases they may
+not be actually independent.  We'll discuss this more in a later
```

Comment:
I think it's a bit less mysterious to say: "though in certain cases they may not be actually independent because of some problem specific constraints."
This gives a first idea that the dofs of the solution GridFunction are not necessarily the same as the dofs (unknowns) of the problem.